### PR TITLE
python: Add 3.10 support

### PR DESCRIPTION
**Pull request by oliver-sanders**
Add support for Python 3.10.

Note 3.11 is now nearing release but we can't install our deps with it just yet...

> **Note:** We may need to change the conda-forge metadata for the 8.1.0 release.

- [x] I have read `CONTRIBUTING.md` and added my name as a Code Contributor.
- [x] Contains logically grouped changes (else tidy your branch by rebase).
- [x] Does not contain off-topic changes (use other PRs for other changes).
- [x] Applied any dependency changes to both `setup.cfg` and `conda-environment.yml`.
- [x] Tests are included (or explain why tests are not needed).
- [x] `CHANGES.md` entry included if this is a change that can affect users
- [x] [Cylc-Doc](https://github.com/cylc/cylc-doc) pull request opened if required at cylc/cylc-doc/pull/XXXX.
- [x] If this is a bug fix, PRs raised to both master and the relevant maintenance branch.

### CODE REVIEW

**Comment by oliver-sanders** [comment]
Blocked by pyuv.

**Comment by ngam** [comment]
> Blocked by pyuv.

does adding libuv as req help?

**Comment by oliver-sanders** [comment]
I think `pyuv` bundles its own libuv.

The issue is that `pyuv` needs some changes to the C code to work with "3.10".

**Comment by dwsutherland** [comment]
Relates to https://github.com/cylc/cylc-uiserver/pull/260

(perhaps why you're upgrading?)

**Comment by oliver-sanders** [comment]
This issue could potentially remove our dependency of pyuv - https://github.com/cylc/cylc-flow/issues/4601

**Comment by oliver-sanders** [comment]
Tried adding `3.11.dev` to the `fast_tests`, however, installation failed - https://github.com/cylc/cylc-flow/runs/7723829198?check_suite_focus=true#step:6:734

Need to wait for our dependencies to release 3.11 wheels.

**Comment by hjoliver** [comment]
> Need to wait for our dependencies to release 3.11 wheels.

I restore the blocked label for now.

**Comment by MetRonnie** [comment]
it doesn't block us from 3.10 support though?

**Comment by hjoliver** [comment]
Sorry, misread!

**Comment by oliver-sanders** [comment]
(rebased)

**Comment by MetRonnie** [comment]
Poking tests

**Review by MetRonnie — APPROVED** [review]
Nice. (Test failure is due to the flaky integration test situation)

**Comment by MetRonnie** [comment]
Changelog entry?

**Comment by oliver-sanders** [comment]
I don't think we need a changelog for this metadata change caused by a dependency of a dependency, it's a bit meta.

**Comment by MetRonnie** [comment]
Thought it might be nice to have "added support for Python 3.10" in the changelog

**Comment by oliver-sanders** [comment]
We didn't really "add" it as much as unblock it, feel free to add if you think it's important.

**Review by hjoliver — APPROVED** [review]
:tada:

**Pull request merged**
